### ISSUE TITLE
Counting transit data twice

### ISSUE THREAD

**Issue by tlacoyodefrijol** [okfn/opendatacensus#774]
@brew Hi, we got this from the US Census people. 

> I referred to "US city open data census" and also looked at each dataset.
> I think some cities' total scores are miscalculated because of doubly
> counted of "transit"data.
> There are two transit data in some cities on entries cascade csv.
> For example, total score of  Chicago is not 1790, it might be 1720 or 1690.
> You might count transit data 100 and 70 again from entries cascade sheet.
> Would you please teach me correct score?

**Comment by brew** [comment]
This is a subtle error when calculating scores for a place from its entries. For some entries the dataset property is 'transit' and others 'transit ' (note the space). It is likely that 'transit ' with-a-space was originally used in the datasets spreadsheet and hence used as the id for entries created for a short period in 2014. This was subsequently corrected to 'transit' without-a-space, and entries then used that. However, when calculating scores these appear as separate sets of entries with different datasets. It looks like 34 records use 'transit ' with-a-space.

I need to:

- [ ] Fix the problematic entry.dataset values by trimming the whitespace in existing entries.
- [ ] Adjust the data loaders to trim when importing values from the google spreadsheets.

**Comment by tlacoyodefrijol** [comment]
![Thanks!](https://media.giphy.com/media/3oEdva9BUHPIs2SkGk/giphy.gif)

**Comment by brew** [comment]
@tlacoyodefrijol This has been fixed on the staging server. All entries with 'transit ' with-a-space have had the space removed. This has changed some of the place scores and rank for a few cities. Can you have the us-city folks look over it to see if the changes are what they expect. http://us-city.next-census.okfn.org/

**Comment by brew** [comment]
There are problems with trailing spaces in other areas too, which may not have an impact on scoring, but I'll try to cover with the import fix and a corresponding db data migration.

- [ ] place.id
- [ ] entry.place
- [ ] entry.dataset
- [ ] dataset.id (no issues here, but import should cover possibility)

**Comment by tlacoyodefrijol** [comment]
@brew that's exactly it!

**Comment by brew** [comment]
@pwalsh Can you think of a reason not to trim whitespace from all inputs in the loader here: https://github.com/okfn/opendatacensus/blob/28fe2e5b949a23043e3ce5849abd96fdd27f130e/census/loaders/utils.js#L59 ?

**Comment by pwalsh** [comment]
Can't think of a reason not to

**Comment by brew** [comment]
@tlacoyodefrijol 

tl;dr I've run a script to remove trailing spaces from a couple of fields. Please check the following sites on staging to ensure they appear as expected. If you're happy, I'll update the production server.

http://us-city.next-census.okfn.org/
http://ar-city.next-census.okfn.org/
http://ie-city.next-census.okfn.org/
http://se-city.next-census.okfn.org/

longer version:

Staging (http://us-city.next-census.okfn.org/) has been updated with the following:

1) A script to trim existing values for entry.place, entry.dataset and place.id. This has been run on staging.

2) Loaders adjusted to remove whitespace during data import, which should stop this happening in the future.

The sites affected were se-city (many place ids had trailing spaces), ie-city (limerick), ar-city (rivadavia), us-city (entry.dataset 'transit')

Handily, fixing `place.id` fixes another error where place pages aren't accessible from links in the site (because they expect a trailing space in the url). For example, the broken page for ar-city/rivadavia (http://ar-city.census.okfn.org/place/rivadavia) is fixed on staging (http://ar-city.next-census.okfn.org/place/rivadavia).

**Comment by brew** [comment]
This is now fixed on the production site.
